src/nvim/getchar.c

```diff
@@ -1806,13 +1806,15 @@ static bool expand_matched_map(mapblock_T *mp, const int keylen, int *mapdepthp)
 // of the mapblock_T m_keylen member in *keylenp.
 // If no mapping is found return NULL.
 //    If this is because the typebuffer contains characters that don't match
-//    any mapping, don't change *keylenp.
+//    any mapping, then leave *keylenp as 0.
 //    If it is because the typebuffer contains the start of a mapping, but not
 //    an entire one, set *keylenp to KEYLEN_PART_MAP.
 static mapblock_T *find_typed_map(const bool timedout, const int local_State,
                                   int *keylenp)
 {
   int temp_c = typebuf.tb_buf[typebuf.tb_off];
+  // Mainly to make it clear this is always 0 when calling this function.
+  assert(*keylenp == 0);
```

Comment:
Perhaps it would be better to document that `*keylenp` will be set to `0` if no (partial) mapping could be found.

I only cursorily looked at the callchain of this function, but it seems like no part of the code relies on the fact that `*keylenp` is unchanged. So making it always changed (and thus always defined) makes the function easier to read.